docs/servo_power_monitor/src/index.ts

```diff
@@ -165,47 +165,47 @@ function setupStartUSBButton() {
 };
 setupStartUSBButton();
 
-requestSerialButton.addEventListener('click', () => {
+var port;
+var reader: ReadableStreamDefaultReader;
+
+requestSerialButton.addEventListener('click', async () => {
   halt = false;
-  navigator.serial
-      .requestPort({filters: [{usbVendorId: 0x18d1, usbProductId: 0x520d}]})
-      .then(async (port) => {
-        // Connect to `port` or add it to the list of available ports.
-        await port.open({baudRate: 115200});
-        const encoder = new TextEncoder();
-        const writer = port.writable.getWriter();
-        await writer.write(encoder.encode('help\n'));
-        writer.releaseLock();
-
-        kickWriteLoop(async (s) => {
-          let data = new TextEncoder().encode(s);
-          const writer = port.writable.getWriter();
-          await writer.write(data);
-          writer.releaseLock();
-        })
-        readLoop(async () => {
-          const reader = port.readable.getReader();
-          try {
-            while (true) {
-              const {value, done} = await reader.read();
-              if (done) {
-                // |reader| has been canceled.
-                break;
-              }
-              return utf8decoder.decode(value);
-            }
-          } catch (error) {
-            console.error(error);
-            throw error;
-          } finally {
-            reader.releaseLock();
-          }
-        });
-      })
-      .catch((e) => {
-        // The user didn't select a port.
-        console.error(e);
-      });
+  requestSerialButton.disabled = true;
```

Comment:
I think it's too early to disable the button, since it's possible that the user cancels the port selection window and the user can't select a port anymore since the button is disabled. how about moving the line after the port selection?

Also, It's a good idea to think what will happen if you disconnect the servo during the measurement is ongoing... I think the button will be kept disabled and the user have to reload the page to restart it again.

Thinking about error case is an important thing ;) (but yeah, it's difficult, so please try!)